### ISSUE TITLE
Can't use .z.ZodArray as output schema

### ISSUE THREAD

**Issue by gmazoni** [RobinTail/express-zod-api#1131]
I can't have a schema like:

```typescript
export const listCoursesSchemaResponse = z
  .object({
    ...
  })
  .array();
```

To be used in a endpoints factory output, an array is a valid json, this is forcing to have an object always.
I'm doing this workaround:

```typescript
export const listCoursesSchemaResponse = z
  .object({
    ...
  })
  .array()  as z.ZodTypeAny;
```
But it would be nice to have ZodArray here: 

```typescript
type IOSchema<U extends UnknownKeysParam = any> = z.ZodObject<any, U> | z.ZodUnion<[IOSchema<U>, ...IOSchema<U>[]]> | z.ZodIntersection<IOSchema<U>, IOSchema<U>> | z.ZodDiscriminatedUnion<string, z.ZodObject<any, U>[]> | Refined<z.ZodObject<any, U>>;
```

**Comment by RobinTail** [comment]
Hello @gmazoni .
This is expected behaviour.
1. Having array on the top level of response schema is a bad practice, consider having the array in an object property.
2. In case you really want to have an array — make a custom ResultHandler that will respond with array. https://github.com/RobinTail/express-zod-api/discussions/361

**Comment by RobinTail** [comment]
@gmazoni :
https://github.com/RobinTail/express-zod-api/discussions/361#discussioncomment-6868815

**Comment by RobinTail** [comment]
> I'm migrating an existing application

I'll make a helper for you soon in #1134

**Comment by RobinTail** [comment]
🚀 The feature that should help you released in v11.7.0, @gmazoni

**Comment by gmazoni** [comment]
WOW! Thanks @RobinTail! Now migrating my application will be a breeze!
